lib/httpd_configmap_generator/ldap.rb

```diff
@@ -0,0 +1,190 @@
+                                      :default     => "member" },
+        :ldap_group_object_class => { :description => "The object class of a group entry in LDAP",
+                                      :default     => "groupOfNames" },
+        :ldap_id_use_start_tls   => { :description => "Must connection also use tls",
+                                      :default     => "true" },
+        :ldap_port               => { :description => "LDAP Directory Port" },
+        :ldap_tls_reqcert        => { :description => "The checks to perform on server certificates.",
+                                      :default     => "allow" },
+        :ldap_user_gid_number    => { :description => "LDAP attribute corresponding to the user's gid",
+                                      :default     => "gidNumber" },
+        :ldap_user_name          => { :description => "LDAP Directory User Name",
+                                      :default     => "cn"},
+        :ldap_user_object_class  => { :description => "Object class of a user entry in LDAP",
+                                      :default     => "posixAccount" },
+        :ldap_user_uid_number    => { :description => "LDAP attribute corresponding to the user's id",
+                                      :default     => "uidNumber" },
+        :ldap_user_search_base   => { :description => "The user DN search scope" },
+        :ldap_group_search_base  => { :description => "The group DN search scope" },
+        :support_non_posix       => { :description => "Suppoert non-posix user recoreds",
+                                      :default     => false },
+      )
+    end
+
+    def persistent_files
+      %w(/etc/nsswitch.conf
+         /etc/openldap/ldap.conf
+         /etc/pam.d/fingerprint-auth-ac
+         /etc/pam.d/httpd-auth
+         /etc/pam.d/password-auth-ac
+         /etc/pam.d/postlogin-ac
+         /etc/pam.d/smartcard-auth-ac
+         /etc/pam.d/system-auth-ac
+         /etc/sssd/sssd.conf
+         /etc/sysconfig/authconfig
+         /etc/sysconfig/network) + [opts[:cert_file]]
+    end
+
+    def configure(opts)
+      update_hostname(opts[:host])
+
+      init_search_base
+      run_auth_config
+      configure_pam
+      configure_sssd
+      chmod_chown_cert_file
+      config_map = ConfigMap.new(opts)
+      config_map.generate(AUTH[:type], realm, persistent_files)
+      config_map.save(opts[:output])
+    rescue => err
+      log_command_error(err)
+      raise err
+    end
+
+    def unconfigure
+      return unless configured?
+      raise "Unable to unconfigure authentication against LDAP"
+    end
+
+    def configured?
+      File.exist?(SSSD_CONFIG)
+    end
+
+    def domain
+      opts[:ldap_basedn].split(",").collect do |p|
+        p.split('dc=')[1]
+      end.compact.join('.')
+    end
+
+    private
+
+    def ldapserver_url
+      opts[:ldap_port] ||=
+        if opts[:ldap_mode].downcase == "ldaps"
+          636
+        else
+          389
+        end
```

Comment:
prefer ternary for above:   opts[:ldap_port] ||= opts[:ldap_mode].downcase == "ldaps" ? 636 : 389

Reply:
rubocop liked it this way. I had it the say you prefer but bubocop complained

Reply:
oh wait, may bad. Got it! Will do:
```ruby
        opts[:ldap_port] ||= opts[:ldap_mode].downcase == "ldaps" ? 636 : 389
        "#{opts[:ldap_mode]}://#{opts[:ldap_host]}:#{opts[:ldap_port]}"
```